pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1472,13 +1523,38 @@ func doAppNetworkModifyUnderlayNetwork(
 	ulConfig *types.UnderlayNetworkConfig,
 	oldulConfig *types.UnderlayNetworkConfig,
 	ulStatus *types.UnderlayNetworkStatus,
-	ipsets []string, force bool) {
+	ipsets []string, intfIdx int, force bool) error {
 
+	netstatus := lookupNetworkInstanceStatus(ctx, ulConfig.Network.String())
+	// release the appNum, on network transition
+	if ulConfig.Network != oldulConfig.Network {
+		appID := status.UUIDandVersion.UUID
+		isStatic := (ulConfig.AppIPAddr != nil)
+		networkID := ulConfig.Network
+		oldNetworkID := oldulConfig.Network
+		oldnetstatus := lookupNetworkInstanceStatus(ctx, oldNetworkID.String())
+		_, err := appNumOnUNetGet(ctx, oldNetworkID, appID, intfIdx)
+		if err == nil {
+			appNumOnUNetFree(ctx, oldNetworkID, appID, intfIdx)
+		}
+		_, err = appNumOnUNetAllocate(ctx, networkID, appID, intfIdx,
+			isStatic, false)
+		if err != nil {
+			addError(ctx, status, "appNetworkModify", err)
+			errStr := fmt.Sprintf("App Num get fail :%s", err)
+			log.Errorf("appNumsOnUNetAllocate(%s, %s, %d): fail: %s",
+				networkID.String(), appID.String(), intfIdx, errStr)
+			return errors.New(errStr)
+		}
+		// TBD:XXX we may need to refactor VLAN
+		// TBD:XXX we may need to refactor Vif
```

Comment:
What do there comments mean?

Reply:
I see you moved the comments down a fine lines, but that doesn't answer my question.